### PR TITLE
[#11878] Upgrade instructor request form UI

### PR DESCRIPTION
**Pull request by xenosf**
Part of #11878

**Outline of Solution**
* Edit existing results page to include new instructor request form
* Set up "I am an instructor" declaration before displaying form
	image
* Create new form component
	image
* Set up initial form validation
	not done yet (to be done in a separate PR): max length validation, form submission status/validation error summary
	image
* Set up submission acknowledgement page
  image
* Add tests for instructor request form component

### CODE REVIEW

File: src/web/app/pages-static/request-page/instructor-request-form/instructor-request-form.component.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  name = this.arf.controls.name;
+  institution = this.arf.controls.institution;
+  country = this.arf.controls.country;
+  email = this.arf.controls.email;
+  homePage = this.arf.controls.homePage;
+  comments = this.arf.controls.comments;
+
+  hasSubmitAttempt = false;
+
+  @Output() requestSubmitted = new EventEmitter<InstructorRequestFormData>();
```
Comment (jayasting98) [inline src/web/app/pages-static/request-page/instructor-request-form/instructor-request-form.component.ts:34]:
```suggestion
  @Output() requestSubmittedEvent = new EventEmitter<InstructorRequestFormData>();
```

I think we usually have `Event` at the end of the name. You can also call it `submitRequestEvent`, or `requestSubmissionEvent`.

File: src/web/app/pages-static/request-page/instructor-request-form/instructor-request-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import { Component, EventEmitter, Output } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { InstructorRequestFormData } from './InstructorRequestFormData';
+
+// eslint-disable-next-line
+const URL_REGEX = /(https?:\/\/)?(www\.)[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,4}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)|(https?:\/\/)?(www\.)?(?!ww)[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,4}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/;
```
Comment (jayasting98) [inline src/web/app/pages-static/request-page/instructor-request-form/instructor-request-form.component.ts:6]: Amazing regex lol 😁

Anyway, we may want to move this to the `FieldValidator` and `ApiConst` in the back-end, and just import this from `src/web/types/form-validator.ts` (you can take a look, there are some existing examples in those files).

Though for now, I guess we can probably just leave this here; we can move it in the PR that will integrate it with the backend.

File: src/web/app/pages-static/request-page/request-page.component.scss [hunk 3]
```diff
@@ -0,0 +1,5 @@
+.empty-field-placeholder::after {
+    content: "(empty)";
+    opacity: 0.5;
+    font-style: italic;
+}
```
Comment (jayasting98) [inline src/web/app/pages-static/request-page/request-page.component.scss:5]: This is a nit: let's add the file's trailing newline.

File: src/web/app/pages-static/request-page/instructor-request-form/instructor-request-form.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    });
+    fixture = TestBed.createComponent(InstructorRequestFormComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+});
```
Comment (jayasting98) [inline src/web/app/pages-static/request-page/instructor-request-form/instructor-request-form.component.spec.ts:21]: Is it possible to test the form component more, like the [output](https://angular.io/guide/testing-components-scenarios#component-with-inputs-and-outputs)?

**Review by jayasting98 — CHANGES_REQUESTED** [review]
Thanks for the work. I have some comments.

**Comment by ziqing26** [comment]
@xenosf remember to address the comments and re-request for review:)

File: src/web/app/pages-static/request-page/instructor-request-form/instructor-request-form.component.scss [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    color: red;
+}
+
+.help-block {
+    margin-bottom: 0.8rem;
+}
+
+.red-font {
+    color: red;
+}
```
Comment (ziqing26) [inline src/web/app/pages-static/request-page/instructor-request-form/instructor-request-form.component.scss:22]: Let's also add a new line at the end of this file

File: src/web/app/pages-static/request-page/instructor-request-form/instructor-request-form.component.ts (hunk 2 above)
Comment (ziqing26) [reply src/web/app/pages-static/request-page/instructor-request-form/instructor-request-form.component.ts:6]: Could we add a small comment explaining what this regex is for? Just for better readability

**Review by ziqing26 — CHANGES_REQUESTED** [review]
Looks good in general, just need to address the few nitpicks we mentioned. Thanks!

File: src/web/app/pages-static/request-page/request-page.component.scss (hunk 3 above)
Comment (xenosf) [reply src/web/app/pages-static/request-page/request-page.component.scss]: Strange, the trailing newline shows up on my end but doesn't appear on GH. I'll try deleting and re-adding the file

Comment (xenosf) [reply src/web/app/pages-static/request-page/request-page.component.scss]: Ok, deleting and re-adding the file seems to have fixed it

**Review by ziqing26 — APPROVED** [review]
LGTM! Thank you for the work!

**Review by jayasting98 — APPROVED** [review]
Great work! Thanks!

**Pull request merged**